radicle-node/src/worker.rs

```diff
@@ -403,6 +355,90 @@ impl<G: Signer + Ecdh + 'static> Worker<G> {
             }
         }
     }
+
+    fn tunnel_fetch<Specs>(
+        &self,
+        repo: &fetch::StagedRepository,
+        specs: Specs,
+        session: WireSession<G>,
+    ) -> (WireSession<G>, Result<(), FetchError>)
+    where
+        Specs: fetch::AsRefspecs,
+    {
+        let mut tunnel = match Tunnel::with(session, net::SocketAddr::from(([0, 0, 0, 0], 0))) {
+            Ok(tunnel) => tunnel,
+            Err((session, err)) => return (session, Err(err.into())),
+        };
+        let res = self._fetch(repo, specs, &mut tunnel);
+        let mut session = tunnel.into_session();
```

Comment:
```suggestion

        let mut session = tunnel.into_session();
```
Just some breathing room to logicallty break up the function